nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.rest.management;
+
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static javax.servlet.http.HttpServletResponse.SC_NO_CONTENT;
+import static org.junit.Assert.assertEquals;
+
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.work.api.WorkManager;
+import org.nuxeo.elasticsearch.ElasticSearchConstants;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.elasticsearch.api.ElasticSearchService;
+import org.nuxeo.elasticsearch.query.NxQueryBuilder;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(AutomationFeature.class)
+@Deploy("org.nuxeo.ecm.core.management")
+@Deploy("org.nuxeo.rest.management.test")
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.elasticsearch.core")
+@Deploy("org.nuxeo.elasticsearch.automation")
+public class TestElasticsearchObject extends ManagementBaseTest {
+
+    public static final String GET_DOCUMENTS_QUERY = "SELECT * from Document";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Inject
+    protected ElasticSearchService ess;
+
+    @Inject
+    protected ElasticSearchAdmin esa;
+
+    @Test
+    public void shouldRunIndexation() throws Exception {
+        // Init indexes and drop if any
+        esa.initIndexes(true);
+
+        // Initial docs count
+        long initialDocCount = coreSession.query(GET_DOCUMENTS_QUERY).totalSize();
+
+        // Nothing indexed because of disable indexing flag
```

Comment:
Nothing indexed because the index was just dropped no?
It lois like this comment makes more sens at line 88, after having created some documents with the `disableAutoIndexing` flag, then waited for the next transaction.